src/coreclr/jit/compiler.cpp

```diff
@@ -2281,7 +2281,15 @@ void Compiler::compSetProcessor()
     }
     if (instructionSetFlags.HasInstructionSet(InstructionSet_AVX512F))
     {
-        instructionSetFlags.AddInstructionSet(InstructionSet_Vector512);
+        if (!DoJitStressEvexEncoding())
+        {
+            instructionSetFlags.RemoveInstructionSet(InstructionSet_AVX512F);
+            // (plus remove rest of AVX512 ISAs)
```

Comment:
I think we need to explicitly remove the rest of them here otherwise the `assert(instructionSetFlags.Equals(EnsureInstructionSetFlagsAreValid(instructionSetFlags)));` might fail.

Reply:
You could do this all at once via:
```cpp
instructionSetFlags = EnsureInstructionSetFlagsAreValid(instructionSetFlags)`
```

Given the `JitStressEvexEncoding` won't exist long term, I think it should be fine. But it does do "extra work", so we wouldn't want to necessarily ship .NET 8 with it doing that.

Reply:
I made the change and relevant failures seem to be fixed. Thank you.

As you pointed out, we'll need to remove the flag by 8.0